src/components/network-area-diagram-viewer/diagram-utils.ts

```diff
@@ -192,3 +192,138 @@ function classIsContainerOfDraggables(element: SVGElement): boolean {
+    const unitaryRadius = vlCircleRadius / (nbNeighbours + 1);
+    return [
+        busIndex * unitaryRadius + interAnnulusSpace / 2,
+        (busIndex + 1) * unitaryRadius - interAnnulusSpace / 2,
+        vlCircleRadius,
+    ];
+}
+
+function getCirclePath(
+    radius: number,
+    angleStart: number,
+    angleEnd: number,
+    clockWise: boolean
+) {
+    const arcAngle = angleEnd - angleStart;
+    const xStart = radius * Math.cos(angleStart);
+    const yStart = radius * Math.sin(angleStart);
+    const xEnd = radius * Math.cos(angleEnd);
+    const yEnd = radius * Math.sin(angleEnd);
+    const largeArc = Math.abs(arcAngle) > Math.PI ? 1 : 0;
+    return (
+        xStart.toFixed(3) +
+        ',' +
+        yStart.toFixed(3) +
+        ' A' +
+        radius.toFixed(3) +
+        ',' +
+        radius.toFixed(3) +
+        ' ' +
+        radToDeg(arcAngle).toFixed(3) +
+        ' ' +
+        largeArc +
+        ' ' +
+        (clockWise ? 1 : 0) +
+        ' ' +
+        xEnd.toFixed(3) +
+        ',' +
+        yEnd.toFixed(3)
+    );
+}
+
+// get path for bus annulus
+export function getFragmentedAnnulusPath(
+    angles: number[],
+    busNodeRadius: [number, number, number],
+    nodeHollowWidth: number
+): string {
+    let path: string = '';
+    const halfWidth = nodeHollowWidth / 2;
+    const deltaAngle0 = halfWidth / busNodeRadius[1];
+    const deltaAngle1 = halfWidth / busNodeRadius[0];
+    for (let index = 0; index < angles.length; index++) {
+        const outerArcStart = angles[index] + deltaAngle0;
+        const outerArcEnd = angles[index + 1] - deltaAngle0;
+        const innerArcStart = angles[index + 1] - deltaAngle1;
+        const innerArcEnd = angles[index] + deltaAngle1;
+        if (outerArcEnd > outerArcStart && innerArcEnd < innerArcStart) {
+            path =
+                path +
+                'M' +
+                getCirclePath(
+                    busNodeRadius[1],
+                    outerArcStart,
+                    outerArcEnd,
+                    true
+                ) +
+                ' L' +
+                getCirclePath(
+                    busNodeRadius[0],
+                    innerArcStart,
+                    innerArcEnd,
+                    false
+                ) +
+                ' Z ';
+        }
+    }
+    return path;
+}
+
+function getPolylinePoints(polylinePoints: string): Point[] | null {
+    const coordinates: string[] = polylinePoints.split(/,| /);
+    if (coordinates.length < 4) {
+        return null;
+    }
+    const points: Point[] = [];
+    for (let index = 0; index < 4; index = index + 2) {
+        const point = new Point(+coordinates[index], +coordinates[index + 1]);
+        points.push(point);
+    }
+    return points;
+}
+
+// get angle of first 2 points of a polyline
+export function getPolylineAngle(polylinePoints: string): number | null {
+    const points: Point[] | null = getPolylinePoints(polylinePoints);
```

Comment:
Similarly for polyline
```suggestion
export function getPolylineAngle(polyline: HTMLElement): number | null {
    const pointsString = polyline.getAttribute('points');
    const points: Point[] | null = getPolylinePoints(pointsString);
```

Reply:
done